File: src/connect4/data/SolverConfig.java
package connect4.data;

import connect4.Solver;

import java.util.Arrays;

public class SolverConfig {
    public int[][] board;
    public int player;
    public int maxTime;
    public int maxDepth;

    public SolverConfig(int[][] board, int player, int maxTime, int maxDepth) {
        this.board = board;
        this.player = player;
        this.maxTime = maxTime;
        this.maxDepth = maxDepth;
    }

    @Override
    public String toString() {
        return "SolverConfig{" +
                "board=" + Arrays.deepToString(board) +
                ", player=" + player +
                ", maxTime=" + maxTime +
                ", maxDepth=" + maxDepth +
                '}';
    }
}

File: webapp/src/wasm/dtos/SolverConfig.ts
export class SolverConfig {
    public board: number[][];
    public player: number;
    public maxTime: number;
    public maxDepth: number;

    constructor(board: number[][], player: number, maxTime: number, maxDepth: number) {
        this.board = board;
        this.player = player;
        this.maxTime = maxTime;
        this.maxDepth = maxDepth;
    }

    public async serialize(lib: any): Promise<any> {
        const SolverConfigLib: any = await lib.connect4.data.SolverConfig;

        return await new SolverConfigLib(this.board, this.player, this.maxTime, this.maxDepth);
    }
}